homeassistant/components/sensor/demo.py

```diff
@@ -4,16 +4,68 @@
 For more details about this platform, please refer to the documentation
 https://home-assistant.io/components/demo/
 """
+import datetime as datetime
+import math
+import random
+import logging
+
+import voluptuous as vol
 from homeassistant.const import ATTR_BATTERY_LEVEL, TEMP_CELSIUS
 from homeassistant.helpers.entity import Entity
+import homeassistant.helpers.config_validation as cv
+from homeassistant.const import CONF_NAME
+from homeassistant.components.sensor import PLATFORM_SCHEMA
+
+_LOGGER = logging.getLogger(__name__)
+SCAN_INTERVAL = datetime.timedelta(seconds=1)
+ICON = 'mdi:chart-line'
+
+CONF_UNIT = 'unit'
+CONF_AMP = 'amplitude'
+CONF_MEAN = 'mean'
+CONF_PERIOD = 'period'
+CONF_PHASE = 'phase'
+CONF_FWHM = 'spread'
+CONF_SEED = 'seed'
+
+DEFAULT_NAME = 'simulated'
+DEFAULT_UNIT = 'value'
+DEFAULT_AMP = 1
+DEFAULT_MEAN = 0
+DEFAULT_PERIOD = 60
+DEFAULT_PHASE = 0
+DEFAULT_FWHM = 0
+DEFAULT_SEED = 999
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_NAME, default=DEFAULT_NAME): cv.string,
+    vol.Optional(CONF_UNIT, default=DEFAULT_UNIT): cv.string,
```

Comment:
Just a suggestion, but I don't understand why we would specify a default unit of measurement of "value". Why not simply make "" (empty string) the default unit? Most other sensors do it that way I think...

Reply:
I found if I left the unit as an empty string, then by default the front end did not plot a graph, but instead one of those barcode-like plots. We could make the default unit % or some other numeric?

Reply:
Oh ok, that's right. Ignore the previous comment then.